lib/samples/apply_class_breaks_renderer_to_sublayer/apply_class_breaks_renderer_to_sublayer_sample.dart

```diff
@@ -28,42 +28,109 @@ class ApplyClassBreaksRendererToSublayerSample extends StatefulWidget {
 class _ApplyClassBreaksRendererToSublayerSampleState
     extends State<ApplyClassBreaksRendererToSublayerSample> {
   final _mapViewController = ArcGISMapView.createController();
+  // create a map with a basemap style
+  final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISStreets);
+  // set the rendered state to false
+  bool _rendered = false;
+  // set the ready state to false
+  bool _ready = false;
+  // create an image sublayer
+  late ArcGISMapImageSublayer countiesSublayer;
+  // create an renderer
+  late Renderer defaultRenderer;
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      body: ArcGISMapView(
-        controllerProvider: () => _mapViewController,
-        onMapViewReady: onMapViewReady,
+      body: SafeArea(
+        top: false,
+        child: Column(
+          // add the map view and buttons to a column.
+          children: [
+            Expanded(
+              // add a map view to the widget tree and set a controller.
+              child: ArcGISMapView(
+                controllerProvider: () => _mapViewController,
+                onMapViewReady: onMapViewReady,
+              ),
+            ),
+            SizedBox(
+              height: 60,
+              // add the buttons to the column.
+              child: Row(
+                crossAxisAlignment: CrossAxisAlignment.center,
+                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                children: [
+                  // apply renderer button
+                  ElevatedButton(
+                    onPressed: !_rendered && _ready ? renderLayer : null,
+                    child: const Text('Change Sublayer Renderer'),
+                  ),
+                  // reset button
+                  ElevatedButton(
+                    onPressed: _rendered ? resetRoutes : null,
+                    child: const Text('Reset'),
+                  ),
```

Comment:
Sorry this is probably my bad for my initial comments about having the default renderer and not getting this fully correct then, there's a lot of sample designs flying around! I've just reviewed the design and readme and there's no mention of a reset button so it looks like we just need the "Change Sublayer Renderer" button to toggle which renderer is being used. I think this also means we can get rid of this `_renderered` boolean which is making things feel more complex than they need to be.

We can have a call to go through this sample if this doesn't make sense.

Reply:
Alright, no worries! It should be easy to change few things around

Reply:
I was thinking it would toggle it but yeah you're right the design doesn't specify that. I guess you will still need the `_renderer` flag then to be able to disable the button once it's renderer.

i.e. you'll still need: `onPressed: _rendered ? changeRenderer : null,`